projects/plugins/boost/app/assets/src/js/pages/getting-started/GettingStarted.svelte

```diff
@@ -1,100 +1,45 @@
 	import { BoostPricingTable } from '../../react-components/BoostPricingTable';
 	import Footer from '../../sections/Footer.svelte';
 	import Header from '../../sections/Header.svelte';
-	import config, { markGetStartedComplete } from '../../stores/config';
-	import { connection } from '../../stores/connection';
+	import { initializeConnection, getUpgradeURL } from '../../stores/connection';
 	import { recordBoostEvent } from '../../utils/analytics';
-	import { getUpgradeURL } from '../../utils/upgrade';
-
-	$: pricing = $config.pricing;
 
 	// svelte-ignore unused-export-let - Ignored values supplied by svelte-navigator.
 	export let navigate, location;
-
-	let initiatingFreePlan = false;
-	let initiatingPaidPlan = false;
-	const dismissedSnackbar = writable( false );
-
-	const snackbarMessage = derived(
-		[ connection, dismissedSnackbar ],
-		( [ $connection, $dismissedSnackbar ] ) => {
-			if ( ! $dismissedSnackbar && ! $connection.connected && $connection.error?.message ) {
-				return $connection.error.message;
-			}
-
-			return null;
-		}
-	);
-
-	/**
-	 * Mark that getting started is completed, and head to the next page.
-	 *
-	 * @param {string} externalPage If specified, head to the external page instead of staying on the dashboard.
-	 */
-	function finishGettingStarted( externalPage?: string ) {
-		markGetStartedComplete();
-
-		if ( externalPage ) {
-			window.location.href = externalPage;
-		} else {
-			navigate( '/', { replace: true } );
-		}
-	}
-
-	/**
-	 * User clicked "Free plan"
-	 */
-	async function chooseFreePlan() {
-		initiatingFreePlan = true;
-
-		try {
-			// Make sure there is a Jetpack connection
-			await connection.initialize();
-
-			// Record this selection. This must be done after the connection is initialized.
-			recordBoostEvent( 'free_cta_from_getting_started_page_in_plugin', {} );
-
-			// Head to the settings page.
-			finishGettingStarted();
-		} catch ( e ) {
-			// Un-dismiss snackbar on error. Actual error comes from connection object.
-			dismissedSnackbar.set( false );
-		} finally {
-			initiatingFreePlan = false;
-		}
+	export let userConnected: boolean;
+	export let pricing: ( typeof Jetpack_Boost )[ 'pricing' ];
+	export let isPremium: boolean;
+	export let domain: string;
+
+	let snackbarMessage: string;
+	let selectedPlan: 'free' | 'premium' | false = false;
+	$: if ( selectedPlan !== false ) {
+		initialize( selectedPlan );
 	}
 
-	/**
-	 * User clicked Premium.
-	 */
-	async function choosePaidPlan() {
-		initiatingPaidPlan = true;
-
+	async function initialize( plan: 'free' | 'premium' ) {
 		try {
 			// Make sure there is a Jetpack connection
-			await connection.initialize();
+			await initializeConnection();
 
 			// Record this selection. This must be done after the connection is initialized.
-			recordBoostEvent( 'premium_cta_from_getting_started_page_in_plugin', {} );
+			recordBoostEvent( `${ plan }_cta_from_getting_started_page_in_plugin`, {} );
```

Comment:
The other day I had to search for events in Boost. Having the full name of an event would be helpful in finding it.

Do you think expanding the comment to include the full names of both versions makes sense here?

Reply:
Fixed